### DISCUSSION TITLE
Clarity required for variants of uncertain length and/or position.

### DISCUSSION THREAD

**Discussion by ifokkema** [HGVSnomenclature/hgvs-nomenclature#205] [HGVS presentation]
An example given in the Q&A of [the DNA deletion](https://hgvs-nomenclature.org/stable/recommendations/DNA/deletion/) page lists `c.(987+123_?)del` and `c.(987+1_?)del`. Its reference sequence isn't given, as the gene isn't named. I believe, however, that this is an invalid description. I'm also a bit confused about implied HGVS nomenclature rules. I've collected some information from the website, and would like the committee to review my conclusions.

Observations:
- The parentheses in the above variant description indicate uncertainty. However, so does the question mark. It's now double uncertain.
- The [page on uncertainty](https://hgvs-nomenclature.org/stable/recommendations/uncertain/) lists that the format `(A_B)_(C_D)` or `(?_B)_(C_?)` can be used for ranges with a certain minimum length, and that `(A_B)` can be used for uncertain ranges. In the latter case, a variant sequence or length is given. Given examples from the uncertainty page:
  - `NC_000023.11:g.(31729716_31774235)_(32216847_32287541)del`
  - `NC_000023.10:g.(?_32238146)_(32984039_?)del`
  - `NC_000003.12:g.(63912602_63912844)delN[15]`
  - `LRG_199t1:c.(71_72)G>A`
- On the same page, this example is given for an insertion in an unknown position: `NC_000002.11:g.?_?ins[NC_000022.10:g.35788169_35788352]`. Note that no parentheses are used around `?_?`.

I draw the current conclusions, partially using rules that I think are implied but not clearly defined:
- Parentheses are used whenever the start or end position of a variant is a range, to clarify what is the variant's start position and what is the variant's end position. E.g., it's unclear what `c.100_200_300del` means, but when parentheses are used, the difference between `c.(100_200)_300del` and `c.100_(200_300)del` is immediately clear.
- Parentheses are used to show a certain given range is uncertain; the variant occurs somewhere in that range. The variant *must* be specified. E.g.,
  - Valid: `c.(100_200)delA`
  - Invalid: `c.100_200delA`
  - Invalid: `c.(100_200)del` (an indication is required of _what_ is deleted, I think)
- These parentheses are not used when two positions are given and one is certain, or both are unknown. E.g.,
  - Valid: `c.?_?insA`
  - Valid: `c.100_?del` (or should this be `c.100_?delN[?]`?)
  - Invalid: `c.100_?delA` (this should be `c.100del`, as it's certain c.100 is involved)
  - Invalid: `c.(100_?)del` (parentheses are redundant)
  - Possibly valid: `c.(100_?)delA` (indicating a deletion of A somewhere at or after c.100)
  - Possibly invalid: `c.100_?delN[(10_20)]` (I believe this should be `c.100_(109_119)del`)

It's unclear to me whether `c.?_?del` is valid. Should that be `c.?del` or `c.?delN[?]` or something else?

I realize I'm digging quite deeply here, but since currently, the HGVS nomenclature website is unclear and shows conflicting examples, I think it's good to resolve this.

**Comment by jfjlaros** [comment]
> Invalid: `c.(100_200)del` (an indication is required of what is deleted, I think)

This could be interpreted as a deletion of length 1 in the region 100 to 200, perhaps detected in a fragment length analysis. Likewiwe, a contraction of length five could be written as `c.(100_200)del5`. A similar interpretation could be used for `c.(100_?)del`, which could then be rewritten as `c.(100_999)del`, assuming `c.999` is the end of the reference sequence.

> Should that be `c.?del` or `c.?delN[?]` [...]

I would interpret the first description as a deletion of one nucleotide. This would make the `N[?]` in the second description redundant.

**Reply by ifokkema** [reply]
> > Invalid: `c.(100_200)del` (an indication is required of what is deleted, I think)
> 
> This could be interpreted as a deletion of length 1 in the region 100 to 200, perhaps detected in a fragment length analysis. 

That would work, the absence of a suffix after "del" being interpreted as a single base deletion. However, I can imagine there are quite a few descriptions out there that likely mean "a deletion with unknown length".
However, if the omission of a suffix would become to assume a length of 1, `c.100_?del` would then become invalid, as it would then indicate a single base deletion surely including position `c.100`.


> Likewiwe, a contraction of length five could be written as `c.(100_200)del5`.

That's not valid syntax; "del5" is invalid and should be written as "delN[5]" if the sequence is unknown (or, alternatively, as "delNNNNN").


> A similar interpretation could be used for `c.(100_?)del`, which could then be rewritten as `c.(100_999)del`, assuming `c.999` is the end of the reference sequence.

That would probably be valid, although I would prefer not to recommend duplicate descriptions for the same variant, especially if they can only be identified as the same description using knowledge of the reference sequence.


> > Should that be `c.?del` or `c.?delN[?]` [...]
> 
> I would interpret the first description as a deletion of one nucleotide. This would make the `N[?]` in the second description redundant.

Not redundant, as they indicate two different things. `c.?del` would then be a single-base deletion, while `c.?delN[?]` indicates a deletion of unknown length.

**Reply by jfjlaros** [reply]
Perhaps we should define what `(a_b)` means first. If it is a point (as it is in the `(a_b)_(c_d)` syntax), then the deletion can only be of length 1 and my suggestion for `(a_b)del5` (or `(a_b)delN[5]` if preferred), would be invalid.

Likewise, I interpret a single `?` as one position, so `?delN[?]` can only be of length 1. Larger deletions I would descibe as `?_?del`.

**Reply by ifokkema** [reply]
> Perhaps we should define what `(a_b)` means first. If it is a point (as it is in the `(a_b)_(c_d)` syntax), then the deletion can only be of length 1 and my suggestion for `(a_b)del5` (or `(a_b)delN[5]` if preferred), would be invalid.

I'm not sure if I follow; `(A_B)` is like a range; e.g., `g.(100_200)delN[50]` is a 50-base deletion anywhere between 100 and 200 (inclusive). So it's not a point, as in, a single position, unless, e.g., `g.(100_200)delA` is given. Or is that not what you mean? I think you mean that the breakpoints are single positions in the `(A_B)_(C_D)` notation? But the positions given are not single points, as in, in `g.100_200del`, the position part `100_200` doesn't describe two points, in my opinion, but a stretch of 101 nucleotides. `100` and `200` are, of course, single points.

(on a related note, there is [inconsistency](https://github.com/HGVSnomenclature/hgvs-nomenclature/discussions/133) in whether A and D are possibly deleted in the `(A_B)_(C_D)del` syntax)

> Likewise, I interpret a single `?` as one position, so `?delN[?]` can only be of length 1. Larger deletions I would descibe as `?_?del`.

That would make sense. The guidelines currently don't describe whether anything MUST follow "del", but all given examples provide a length or sequence.

**Reply by jfjlaros** [reply]
> I think you mean that the breakpoints are single positions in the `(A_B)_(C_D)` notation?

Indeed, but if `(A_B)` is a range when used separately, my original interpretations may be valid.

**Reply by ifokkema** [reply]
OK, so then I think I have:

- Valid: `c.(100_200)delA`
- Invalid: `c.100_200delA`
- Valid: `c.(100_200)del` (a single-base deletion within this range, so shorthand for `c.(100_200)delN`)
- Valid: `c.?_?insA`
- Invalid: `c.100_?del` ("del" is a single-base deletion, so this would be `c.100del`)
- Invalid: `c.100_?delA` (for the same reason)
- Valid: `c.(100_?)del` (parentheses indicate an uncertain range, so this is a single base deletion between c.100 and the end of the reference sequence)
- Valid: `c.(100_?)delA` (indicating a deletion of A somewhere at or after c.100)
- Valid: `c.?del` (a single-base deletion at an unknown location)

But now, looking at this, I realize that interpreting `c.?_?del` as a multi-base deletion at an unknown location, in my opinion, conflicts with `c.(100_200)del` being a single-base deletion. That would mean that a multi-base deletion will need to become `c.?_?delN[?]`, i.e., it needs something after "del". Right?

And then we still need to clarify whether `c.100_?delN[(10_20)]` is valid or whether the notation `c.100_(109_119)del` is preferred/required.

**Reply by jfjlaros** [reply]
Yes that seems about right to me.

> [...] a multi-base deletion will need to become `c.?_?delN[?]`, i.e., it needs something after "del". Right?

If I look at the interpretation of a normal deletion `A_Bdel`, both the start and the end of the deletion are given, allowing us to deduce its size. Analogous to this, I interpret `?_?del` as a deletion where both the start and the end are unknown, which makes the size unknown as well. I see no reason to make this more explicit.

`(A_B)del` could perhaps also be interpreted as any deletion between `A` and `B`. If additionally the size of this deletion is known, it can be added after the "del".

**Reply by ifokkema** [reply]
> Yes that seems about right to me.
> 
> > [...] a multi-base deletion will need to become `c.?_?delN[?]`, i.e., it needs something after "del". Right?
> 
> If I look at the interpretation of a normal deletion `A_Bdel`, both the start and the end of the deletion are given, allowing us to deduce its size. Analogous to this, I interpret `?_?del` as a deletion where both the start and the end are unknown, which makes the size unknown as well. I see no reason to make this more explicit.

You suggested `c.(100_200)del` to be a deletion of one base, while `c.(100_200)delN[5]` would be a deletion of five bases. Following that definition, `c.?_?del` would be, in my opinion, confused with a deletion of one base. But since indeed `c.100_200del` is a deletion of 101 bases, perhaps defining `c.(100_200)del` as a single-base deletion is confusing as well.


> `(A_B)del` could perhaps also be interpreted as any deletion between `A` and `B`. If additionally the size of this deletion is known, it can be added after the "del".

Perhaps that's a better idea, indeed.

**Comment by jtdendunnen** [comment]
In my opinion

Valid formats are;

- g.(100_200)delN and g.(100_200)delN[7] and g.(100_200)delN[(10_20)]

	g.100_200delA can not be valid. When you know there is an A deleted you 
have sequenced the fragment so you must also know the position of the A.
	g.?del can not be valid. You have seen a fragment which is 1 nucleotide 
shorter, the fragment has specific positions so you should use 
g.(A_B)delN where A and B specify the positions of the fragment.

- g.(?_100)_(200_?)del
this follows the rule that four positions should be used for deletions 
where the breakpoints are not sequenced but where the minimal size is known.

	Invalid are g.100_?del, g.100_?delA, g.(100_?)del, etc. Depending on 
what you want to report I can see either
	- g.(?_100)_(100_?)del to indicate position 100 is deleted but you do 
not know how far it exentds in either direction.
	- or g.100_(100_?)del to indicate the deletion affects only nucleotides 
donwstream of nucleotide 100.


- g.?_?ins[NC_xxxxxx.y:g.A_B], i.e. you found an extra copy of the 
sequence NC_xxxxxx.y:g.A_B inserted at an unknown position in the 
genome. NC_000023.11:g.?_?ins[NC_xxxxxx.y:g.A_B would mean the sequence 
is inserted at an unknown position in NC_000023.11 (chromosome X). 
g.?_?ins[NC_xxxxxx.y:g.(?_A)_(B_?] means you do not know the exact 
extent of the inserted sequence.

	g.?_?insA can not be valid. When you know there is an A inserted you 
have sequenced the fragment so you must also know the flanking positions.

**Reply by jfjlaros** [reply]
I think we should careful with labeling descriptions as invalid. Instead, we should be lenient as long as a description does not contain internal inconsistencies and can be interpreted unambiguously. In my opinion, the nomenclature may also be used to express hypothetical variants. Otherwise, we only create more exceptions, more rules, more complexity and thereby less acceptation of the nomenclature.

Furthermore, labeling interpretable descriptions as invalid is not useful because these descriptions still need to normalised. We have to assign a meaning to these types of descriptions. I like to see this discussion as an exercise to identify truly invalid (internally inconsistent) descriptions and to resolve ambiguities by assigning meanings that are consistent with the current rules.

For example,

> `g.?del` can not be valid. You have seen a fragment which is 1 nucleotide shorter, the fragment has specific positions [...]

Perhaps we have identified a shorter fragment in a pool of fragments of the same length by means other than sequencing.

> `g.?_?insA` can not be valid. When you know there is an A inserted you have sequenced the fragment so you must also know the flanking positions.

Knowing the insertion does not necessarily mean the sample was sequenced. I am thinking about insertions as a result of retroviral activity or some artificial means like the one used in the Nextera protocol. Additionally, this statement seems inconsistent with allowing `?_?ins[NC_xxxxxx.y:g.A_B]`.

**Reply by ifokkema** [reply]
@jtdendunnen 
> - or g.100_(100_?)del to indicate the deletion affects only nucleotides donwstream of nucleotide 100.

You mean the deletion affects nucleotide 100 *and* possibly nucleotides downstream, right? If it's unclear whether nucleotide 100 is affected as well, the logical notation, in my opinion, would be `g.(100_?)del`.


> - g.?_?ins[NC_xxxxxx.y:g.A_B], i.e. you found an extra copy of the sequence NC_xxxxxx.y:g.A_B inserted at an unknown position in the genome.

That's problematic; variants need reference sequences, so the `g.?_?` needs a reference sequence as well. This is impossible without indicating a chromosome as well. Another issue related to databases is that the informative information for users is actually within the insertion, and not the `g.?_?`. E.g., in LOVD, this insertion would sort all the way to the beginning of the view, whereas users might expect it sorted around other A_B variants. If the location of insertion is completely unknown, perhaps it would be good to have a CNV-like notation, which will also help in general with duplications for which there is no proof that they are in tandem (e.g., `NC_xxxxxx.y:g.A_B[3]` or `NC_xxxxxx.y:g.A_B{3}`).


@jfjlaros 
> I think we should careful with labeling descriptions as invalid. Instead, we should be lenient as long as a description does not contain internal inconsistencies and can be interpreted unambiguously. In my opinion, the nomenclature may also be used to express hypothetical variants. Otherwise, we only create more exceptions, more rules, more complexity and thereby less acceptation of the nomenclature.

I agree with this, also because variant detection techniques will change, and basing requirements on what kind of detection is available right now can cause problems with both old and new data.


> Furthermore, labeling interpretable descriptions as invalid is not useful because these descriptions still need to normalised. We have to assign a meaning to these types of descriptions. I like to see this discussion as an exercise to identify truly invalid (internally inconsistent) descriptions and to resolve ambiguities by assigning meanings that are consistent with the current rules.

That's actually exactly what I'm doing this research for; I'm trying to interpret all kinds of variants, even with "invalid" or "alternative" notations, and then normalize them. When there are multiple notations for the same variant, e.g., `c.100_?delN[(10_20)]` vs `c.100_(109_119)del`, I would like to know which one is "correct"/preferable, so I know what to actually use/present.


> Knowing the insertion does not necessarily mean the sample was sequenced. I am thinking about insertions as a result of retroviral activity or some artificial means like the one used in the Nextera protocol. Additionally, this statement seems inconsistent with allowing `?_?ins[NC_xxxxxx.y:g.A_B]`.

Although I agree with you, it's not really inconsistent; once `A_B` is large enough, one can determine that this sequence is unique to the human genome and found only in that place. If, e.g., by using probes, this sequence is detected to be present more than twice, some report a duplication, which isn't necessarily correct as it's unknown whether this is a tandem duplication. Others would use syntax like `NC_xxxxxx.y:g.A_B[3]` or `NC_xxxxxx.y:g.A_B{3}` to indicate this sequence is found three times in total. However, when somehow an "A" was detected to be inserted somewhere, one is currently assumed to have sequenced flanking regions, simply because "A" is nowhere near unique in the human genome.

**Reply by jfjlaros** [reply]
> Although I agree with you, it's not really inconsistent; once A_B is large enough, [...]

I was specifically thinking about using this notation to insert very small sequences.

**Comment by jtdendunnen** [comment]
the description c.?_?delN[?] is invalid. You can not detect a deletion 
and have no information about the sequences that were detected to be 
deleted. What can be the case is that you have not sequenced the break 
point and do not know the exact extent of the deletion.
